apps/comps/rpc/router/agent/create-agent.ts

```diff
@@ -0,0 +1,63 @@
+import { ORPCError } from "@orpc/server";
+
+import { ApiError, CreateAgentBodySchema } from "@recallnet/services/types";
+
+import { base } from "@/rpc/context/base";
+import { authMiddleware } from "@/rpc/middleware/auth";
+import { serializeAgentPublic } from "@/rpc/router/utils/serialize-agent";
+
+export const createAgent = base
+  .use(authMiddleware)
+  .input(CreateAgentBodySchema)
+  .handler(async ({ input, context, errors }) => {
+    try {
+      const agent = await context.agentService.createAgent({
+        ownerId: context.user.id,
+        name: input.name,
+        handle: input.handle,
+        description: input.description,
+        imageUrl: input.imageUrl,
+        metadata: input.metadata,
+        email: input.email,
+      });
+
+      // Return sanitized and serialized agent
+      const sanitizedAgent = {
+        ...context.agentService.sanitizeAgent(agent),
+        email: agent.email,
+        deactivationReason: agent.deactivationReason,
+        deactivationDate: agent.deactivationDate,
+      };
+
+      return {
+        agent: serializeAgentPublic(sanitizedAgent),
+      };
+    } catch (error) {
+      // Re-throw if already an oRPC error
+      if (error instanceof ORPCError) {
+        throw error;
+      }
```

Comment:
I don't understand the need for this. I can see in the above try block there is no ORPCError thrown. Is this just future proofing?

Reply:
pretty sure there was at least one of these that did throw an ORPCError internally which is how the first one got this.  After that it probably just became a pattern that got copied.  Probably not really necessary for this function, but I do kind of think its nice future-proofing.